### PR TITLE
fix(typing): 'axios' import in typings

### PR DESCRIPTION
**Pull request by geigerzaehler**
We correctly import `axios` in the type definitions `index.d.ts`.

To ensure that typings are checked on CI we run `npm run test-typescript-declaration`. This requires us to upgrade typescript to the latest version.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import axios from 'axios'
+import * as axios from 'axios'
```
Comment (Khaledgarbaya) [inline index.d.ts]: I might be wrong but this might not necessary since we are using `esModuleInterop`

Comment (geigerzaehler) [reply index.d.ts]: If you revert this change and run `tsc --strict index.d.ts` the compiler will complain about `axios` not being a namespace. This is also why I added the typechecking to CI. `esModuleInterop` only changes how code is imported at runtime.

**Comment by geigerzaehler** [comment]
@Khaledgarbaya Any update on this?

**Pull request merged**